src/App.tsx

```diff
@@ -1,173 +1,72 @@
-                      href="#/people/emma-de-milliano-1876"
-                    >
-                      Emma de Milliano
-                    </a>
-                  </td>
-
-                  <td>
-                    <a href="#/people/emile-haverbeke-1877">
-                      Emile Haverbeke
-                    </a>
-                  </td>
-                </tr>
-
-                <tr data-cy="person" className="has-background-warning">
-                  <td>
-                    <a href="#/people/jan-frans-van-brussel-1761">
-                      Jan Frans van Brussel
-                    </a>
-                  </td>
-
-                  <td>m</td>
-                  <td>1761</td>
-                  <td>1833</td>
-                  <td>-</td>
-
-                  <td>
-                    <a href="#/people/jacobus-bernardus-van-brussel-1736">
-                      Jacobus Bernardus van Brussel
-                    </a>
-                  </td>
-                </tr>
-
-                <tr data-cy="person">
-                  <td>
-                    <a
-                      className="has-text-danger"
-                      href="#/people/lievijne-jans-1542"
-                    >
-                      Lievijne Jans
-                    </a>
-                  </td>
-
-                  <td>f</td>
-                  <td>1542</td>
-                  <td>1582</td>
-                  <td>-</td>
-                  <td>-</td>
-                </tr>
-
-                <tr data-cy="person">
-                  <td>
-                    <a href="#/people/bernardus-de-causmaecker-1721">
-                      Bernardus de Causmaecker
-                    </a>
-                  </td>
-
-                  <td>m</td>
-                  <td>1721</td>
-                  <td>1789</td>
-
-                  <td>
-                    <a
-                      className="has-text-danger"
-                      href="#/people/livina-haverbeke-1692"
-                    >
-                      Livina Haverbeke
-                    </a>
-                  </td>
-
-                  <td>
-                    <a href="#/people/lieven-de-causmaecker-1696">
-                      Lieven de Causmaecker
-                    </a>
-                  </td>
-                </tr>
-              </tbody>
-            </table>
+import { Navigate, Route, Routes } from 'react-router-dom';
+import { People } from './pages/People';
+import { Person } from './types';
+import { getPeople } from './api';
+import { PageNavLink } from './components/PageNavLink';
+import { NotFound } from './pages/NotFound';
+import { Home } from './pages/Home';
+
+export const App = () => {
+  const [peopleFromServer, setPeopleFromServer] = useState<Person[]>();
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  useEffect(() => {
+    getPeople()
+      .then((persons) => {
+        setPeopleFromServer(persons);
+        if (peopleFromServer?.length === 0) {
```

Comment:
```suggestion
        if (!peopleFromServer.length) {
```

Reply:
Here I check not the existence of the array, but its content. This code will throw an error if an empty array comes from the server. Which means I can't, as he suggests further, write an empty array by default.

Reply:
More I answer in chat.

Reply:
Your logic is not readable for other developers, you can create additional state to catch errors. isError, for example